WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift

```diff
@@ -0,0 +1,220 @@
+import Foundation
+
+public class ThemeBrowserViewController : UICollectionViewController, UICollectionViewDelegateFlowLayout, NSFetchedResultsControllerDelegate, UISearchBarDelegate {
+    
+    // MARK: - Properties: must be set by parent
+    
+    /**
+     *  @brief      The blog this VC will work with.
+     *  @details    Must be set by the creator of this VC.
+     */
+    public var blog : Blog!
+    
+    // MARK: - Properties
+    
+    /**
+     *  @brief      The FRC this VC will use to display filtered content.
+     */
+    private lazy var themesController: NSFetchedResultsController = {
+        let fetchRequest = NSFetchRequest(entityName: Theme.entityName())
+        fetchRequest.fetchBatchSize = 20
+        let predicate = NSPredicate(format: "blog == %@", self.blog)
+        fetchRequest.predicate = predicate
+        let sort = NSSortDescriptor(key: "order", ascending: true)
+        fetchRequest.sortDescriptors = [sort]
+        let frc = NSFetchedResultsController(fetchRequest: fetchRequest,
+            managedObjectContext: self.managedObjectContext,
+            sectionNameKeyPath: nil,
+            cacheName: nil)
+        frc.delegate = self
+        
+        do {
+            try frc.performFetch()
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching themes: \(error.localizedDescription)")
+        }
+        
+        return frc
+    }()
+    
+    /**
+     *  @brief      The themes service we'll use in this VC and its helpers
+     */
+    private lazy var themeService : ThemeService = {
+        ThemeService(managedObjectContext: self.managedObjectContext)
+    }()
+    private lazy var managedObjectContext = {
+        ContextManager.sharedInstance().mainContext
+    }()
+    private var retiredThemes = Set<Theme>()
+    private var fetchAnimation = false
+   
+    // MARK: - UIViewController
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        title = NSLocalizedString("Themes", comment: "Title of Themes browser page")
+        
+        WPStyleGuide.configureColorsForView(view, collectionView:collectionView)
+        
+        updateThemes()
+    }
+
+    public override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+        
+        collectionView?.collectionViewLayout.invalidateLayout()
+    }
+
+    // MARK: - Updating the list of themes
+    
+    private func updateThemes() {
+        if let fetchedThemes = themesController.fetchedObjects as? [Theme] where !fetchedThemes.isEmpty {
+            retiredThemes = retiredThemes.union(Set(fetchedThemes))
+        } else {
+            fetchAnimation = true
```

Comment:
(Maybe) encapsulate this `fetchAnimation` + WPNoResultsView into a "Display Spinner" / "Hide Spinner" method, for readability?